src/main/java/uk/gov/pay/directdebit/events/dao/GoCardlessEventDao.java

```diff
@@ -65,6 +72,40 @@
     @GetGeneratedKeys
     Long insert(@BindBean GoCardlessEvent goCardlessEvent);
 
-    @SqlUpdate("UPDATE gocardless_events t SET internal_event_id = :eventId WHERE t.id = :id")
+    @SqlUpdate("UPDATE gocardless_events SET internal_event_id = :eventId WHERE id = :id")
     int updateEventId(@Bind("id") Long id, @Bind("eventId") Long eventId);
+    
+    @SqlQuery("SELECT id, " +
+            "internal_event_id, " +
+            "event_id, " +
+            "action, " +
+            "created_at, " +
+            "details_cause, " +
+            "details_description, " +
+            "details_origin, " +
+            "details_reason_code, " +
+            "details_scheme, " +
+            "resource_type," +
+            "links_mandate, " +
+            "links_new_customer_bank_account, " +
+            "links_new_mandate, " +
+            "links_organisation, " +
+            "links_parent_event, " +
+            "links_payment, " +
+            "links_payout, " +
+            "links_previous_customer_bank_account, " +
+            "links_refund, " +
+            "links_subscription, " +
+            "json, " +
+            "event_id " +
+            "FROM gocardless_events " +
+            "WHERE links_mandate = :linksMandate " +
+            "AND links_organisation = :linksOrganisation " +
+            "AND action IN (<applicableActions>) " +
+            "ORDER BY created_at DESC " +
+            "LIMIT 1")
+    Optional<GoCardlessEvent> findLatestApplicableEventForMandate(@Bind("linksMandate") GoCardlessMandateId goCardlessMandateId,
+                                                                        @Bind("linksOrganisation") GoCardlessOrganisationId goCardlessOrganisationId,
```

Comment:
I didn’t really want to create `GoCardlessMandateIdAndOrganisationId` in the first place; I only did it because I needed an object that could encapsulate both for insertion into a `Set`

Reply:
Ah I see, I don't feel strongly, my suggestion was only based on `GoCardlessMandateIdAndOrganisationId` being available and it reducing the number of args to the method. I'm happy if you don't wish to use it here.